pynxtools/dataconverter/helpers.py

```diff
@@ -242,7 +242,7 @@ def is_value_valid_element_of_enum(value, elem) -> Tuple[bool, list]:
     "NX_DATE_TIME": (str,),
     "NX_FLOAT": (float, np.ndarray, np.floating),
     "NX_INT": (int, np.ndarray, np.signedinteger),
-    "NX_UINT": (np.ndarray, np.unsignedinteger),
+    "NX_UINT": (int, np.ndarray, np.unsignedinteger),
```

Comment:
This is not compatible as a Python int is either an C int or long if I'm not wrong. POSINT should suffice if you want a definitely positive number. UINT is more about data representation and increasing the range of the possible numbers that can be represented.  If you are actually reading in a UINT, then I would prefer you bring it as is here for validation.

Or is there something else I'm missing here?

Reply:
No, I actually just followed what was written in NXopt, which just requires that its a positive number. I would just change it to NX_POSINT in the appdef and revert the changes here.